### DISCUSSION TITLE
Error when installing MOAB as DAGMC dependency

### DISCUSSION THREAD

**Discussion by alyaajsmn** [svalinn/DAGMC#954] [General]
Hi,

I'm trying to install DAGMC as a dependency of OpenMC. I'm using the instructions [here](https://svalinn.github.io/DAGMC/install/dependencies.html) to install the dependencies for DAGMC. I encountered some problems when installing MOAB, specifically after running '`autoreconf -fi`', I obtained the following error:

`error: required file 'config/MOABConfig.h.in' not found`

However, this above file is not contained in the [source directory](https://bitbucket.org/fathomteam/moab)

Later on I tried to install DAGMC, and ran into an error after running this:
```
$ cmake ../src -DMOAB_DIR=$HOME/dagmc_bld/MOAB \
               -DBUILD_TALLY=ON \
               -DCMAKE_INSTALL_PREFIX=$INSTALL_PATH
```
              
The error: 
```
-- Submodule update

-- CMAKE_BUILD_TYPE not specified, defaulting to Release
-- CMAKE_BUILD_TYPE: Release


CMake Error at cmake/FindMOAB.cmake:17 (message):
  Could not find MOAB.  Set -DMOAB_DIR=<MOAB_DIR> when running cmake or use
  the $MOAB_DIR environment variable.
Call Stack (most recent call first):
  CMakeLists.txt:62 (find_package)


-- Configuring incomplete, errors occurred!
```

How can I fix this issue?

**Comment by ahnaf-tahmid-chowdhury** [comment]
Hello, `DMOAB_DIR` is the path where you have installed the MOAB project. You need to install [MOAB](https://bitbucket.org/fathomteam/moab/) first, and then you can install DAGMC. We are currently updating our installation methods and hope to soon provide an automated process that installs the required dependencies while installing DAGMC.

**Comment by fbsmith** [comment]
I'm using the same instructions to install MOAB. For me, I run `autoreconf -fi` and I get this error

`(base) fbsmith@Ada:~/dagmc/MOAB/moab$ autoreconf -fi`
error messages...
`configure.ac:53: warning: The macro "AC_ERROR" is obsolete.`
`configure.ac:53: You should run autoupdate.`
error messages...
`configure.ac:112: warning: The macro "AC_LANG_CPLUSPLUS" is obsolete.`
`configure.ac:112: You should run autoupdate.`
error messages...
`configure.ac:112: warning: The macro "AC_TRY_COMPILE" is obsolete.`
`configure.ac:112: You should run autoupdate.`
error messages...
`configure.ac:91: error: possibly undefined macro: AC_DISABLE_SHARED
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
autoreconf: error: /usr/bin/autoconf failed with exit status: 1`

I run `autoupdate` as it recommends, but to no avail. What should be my next step?

**Reply by makeclean** [reply]
So when you run autoupdate, and then re-run autoreconf -fi you get the same error again?

**Reply by fbsmith** [reply]
Yes, that is correct

**Reply by ahnaf-tahmid-chowdhury** [reply]
Make sure you have installed necessary build tools like: autoconf, automake and libtool

```sh
sudo apt-get install autoconf automake libtool #for debian

```
Or you can build MOAB using CMake as well

```cmake
cd build
cmake .. \
        -DENABLE_HDF5=ON \
        -DHDF5_ROOT=${HDF5_INSTALL_DIR} \
        -DCMAKE_INSTALL_PREFIX=${MOAB_INSTALL_DIR} \
        -DENABLE_BLASLAPACK=OFF \
        -DBUILD_SHARED_LIBS=ON \
        -DENABLE_FORTRAN=OFF \
        -DENABLE_PYMOAB=OFF
 make
 make install
```

**Reply by youyeonc** [reply]
Hello, I tried to build MOAB as explained in https://svalinn.github.io/DAGMC/install/dependencies.html but I got "configure: error: LAPACK library not found" error and couldn't proceed. 

Instead, I built MOAB as @ahnaf-tahmid-chowdhury suggested above and it seemed to be installed successfully as "which mbconvert" command worked. But now I'm getting the following error when I try to install DAGMC. Any thoughts?

```
yychoi@yychoi-Precision-7920-Tower:~/dagmc_bld/DAGMC/bld$ cmake ../src -DMOAB_DIR=$HOME/dagmc_bld/MOAB \
               -DBUILD_TALLY=ON \
               -DCMAKE_INSTALL_PREFIX=$INSTALL_PATH

-- CMAKE_BUILD_TYPE not specified, defaulting to Release
-- CMAKE_BUILD_TYPE: Release


CMake Error at cmake/FindMOAB.cmake:19 (message):
  Could not find MOAB.  Set -DMOAB_DIR=<MOAB_DIR> when running cmake or use
  the $MOAB_DIR environment variable.
Call Stack (most recent call first):
  CMakeLists.txt:60 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/yychoi/dagmc_bld/DAGMC/bld/CMakeFiles/CMakeOutput.log".
```
